src/Thunks/api-ms-win-core-processthreads.hpp

```diff
@@ -158,6 +158,44 @@ namespace YY
 #endif
 
 
+#if (YY_Thunks_Support_Version < NTDDI_WS03)
+
+		//Windows Vista, Windows XP Professional x64 Edition [desktop apps only]
+		//Windows Server 2008, Windows Server 2003 with SP1 [desktop apps only]
+		__DEFINE_THUNK(
+		kernel32,
+		4,
+		BOOL,
+		WINAPI,
+		SetThreadStackGuarantee,
+			_Inout_ ULONG *StackSizeInBytes
+		)
+		{
+			if (auto pSetThreadStackGuarantee = try_get_SetThreadStackGuarantee())
+			{
+				return pSetThreadStackGuarantee(StackSizeInBytes);
+			}
+			else
+			{
+				ULONG prev_size = ((TEB*)NtCurrentTeb())->GuaranteedStackBytes;
+				ULONG new_size = (*StackSizeInBytes + 4095) & ~4095;
```

Comment:
`4095 = 4096 - 1` ? 
and in Windows 7.   it's
```cpp
(*StackSizeInBytes + GetSystemInfo().dwPageSize * 2 - 1) & ~(GetSystemInfo().dwPageSize - 1);
```

Reply:
Can you tell me more useful information?

Reply:
@mingkuang-Chuyu I have read the implementation from WoW64's kernel32.dll in Windows Server 2003 Enterprise x64 with Service Pack 2. I found the similar logic from your said in Windows 7.